runtime/autoload/man.vim

```diff
@@ -11,127 +22,207 @@ catch /E145:/
+  else
+    return [ret[0], '']
   endif
+endfunction
 
-  silent exec editcmd.' man://'.page.(empty(sect)?'':'('.sect.')')
+" parses the page and sect out of 'path/page.sect'
+function! s:parse_page_and_sect_path(path) abort
+  let tail = fnamemodify(a:path, ':t')
+  if fnamemodify(tail, ':e') =~# s:man_extensions
+    let tail = fnamemodify(tail, ':r')
+  endif
+  " TODO(nhooyr) all substitutes to matchlist/matchstr maybe?
+  let page = substitute(tail, '^\(\f\+\)\..\+$', '\1', '')
+  let sect = substitute(tail, '^\f\+\.\(.\+\)$', '\1', '')
+  return [page, sect]
+endfunction
 
+function! s:read_page(sect, page, cmd)
+  silent execute s:find_man(a:cmd) 'man://'.a:page.(empty(a:sect)?'':'('.a:sect.')')
   setlocal modifiable
-  silent keepjumps norm! 1G"_dG
-  if empty($MANWIDTH)
-    let $MANWIDTH = winwidth(0)
-  endif
-  silent exec 'r!/usr/bin/man '.s:cmd(sect, page).' | col -b'
-  " Remove blank lines from top and bottom.
+  " remove all the text, incase we already loaded the manpage before
+  silent keepjumps %delete _
+  let $MANWIDTH = winwidth(0)-1
+  " read manpage into buffer
+  silent execute 'r!'.s:man_cmd.s:man_args(a:sect, a:page)
+  call man#normalize_page()
+  setlocal filetype=man
+endfunction
+
+function! man#normalize_page()
+  " remove all those backspaces
+  execute "silent keepjumps %substitute,.\b,,g"
+  " remove blank lines from top and bottom.
   while getline(1) =~# '^\s*$'
     silent keepjumps 1delete _
   endwhile
+  " TODO(nhooyr) is deleting the bottom lines necessary?
+  " I think only deleting the first line is necessary when using r! to
+  " read manpage in.
   while getline('$') =~# '^\s*$'
     silent keepjumps $delete _
   endwhile
-  setlocal nomodified
-  setlocal filetype=man
+  keepjumps 1
+endfunction
 
-  if invoked_from_man || editcmd ==# 'tabnew'
-    call s:set_window_local_options()
+function! s:man_args(sect, page) abort
+  if !empty(a:sect)
+    return s:man_sect_arg.' '.shellescape(a:sect).' '.shellescape(a:page)
   endif
+  return shellescape(a:page)
 endfunction
 
-function s:set_window_local_options() abort
-  setlocal colorcolumn=0 foldcolumn=0 nonumber
-  setlocal nolist norelativenumber nofoldenable
+function! s:error(msg) abort
+  redrawstatus!
+  echon 'man.vim: '
+  echohl ErrorMsg
+  echon a:msg
+  echohl None
 endfunction
 
-function man#pop_page() abort
-  if s:man_tag_depth > 0
-    let s:man_tag_depth = s:man_tag_depth - 1
-    exec "let s:man_tag_buf=s:man_tag_buf_".s:man_tag_depth
-    exec "let s:man_tag_lin=s:man_tag_lin_".s:man_tag_depth
-    exec "let s:man_tag_col=s:man_tag_col_".s:man_tag_depth
-    exec s:man_tag_buf."b"
-    exec s:man_tag_lin
-    exec "norm! ".s:man_tag_col."|"
-    exec "unlet s:man_tag_buf_".s:man_tag_depth
-    exec "unlet s:man_tag_lin_".s:man_tag_depth
-    exec "unlet s:man_tag_col_".s:man_tag_depth
-    unlet s:man_tag_buf s:man_tag_lin s:man_tag_col
+function! man#complete(ArgLead, CmdLine, CursorPos) abort
```

Comment:
I wouldn't call it standard.  Only found two instances of those arguments.  One in `map.txt` and one in `sql.vim`.  IMO, source consistency trumps standards if it doesn't stop code execution.  I'm not that hung up on it though 🙂